proposals/0000-package-dependency-import-syntax.md

```diff
@@ -0,0 +1,155 @@
+
+This proposal would dramatically improve the developer experience for writing scripts in Swift. Dependencies for common use-cases would be easy to integrate and scripts would be portable across machines without additional tools installed on the host machine. 
+
+## Proposed solution
+
+We propose to add a new attribute @package to the import statement for referencing Swift packages. This attribute would be only valid when running in the Swift interpreter (via `swift MyScript.swift` or `swift run MyScript.swift`) and will be rejected if used during compilation of a regular Swift module. Swift Package Manager will then fetch and build the declared packages and make them available to the script when executing it.
+
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+This script can be executable by either `swift MyScript.swift` or `swift run MyScript.swift` which will utilize Swift package managers dependency fetching and resolution logic.
+
+The `@package(...)`syntax will allow the same parameters as SwiftPM's [Package Dependency Description](https://docs.swift.org/package-manager/PackageDescription/PackageDescription.html#package-dependency) which should feel natural to developers familiar with Swift Package Manager manifest syntax.  
+
+## Detailed design
+
+
+#### Basic usage
+The most common case is expected to be importing a dependency where the product name is equivalent to the imported module name.
+```swift
+@package(url: "https://github.com/jpsim/Yams.git", from: "2.0.0")
+import Yams
+```
+
+When a dependency product and target names do not match, the product must be explicitly specified.
+
+For an example, lets look at the Package description below for `swift-tools-support-core`. 
+```swift
+let package = Package(
+    name: "swift-tools-support-core",
+    products: [
+        .library(
+            name: "SwiftToolsSupport",
+            type: .dynamic,
+            targets: ["TSCBasic", "TSCUtility"]),
+        .library(
+            name: "SwiftToolsSupport-auto",
+            targets: ["TSCBasic", "TSCUtility"]),
+
+        .library(
+            name: "TSCTestSupport",
+            targets: ["TSCTestSupport"]),
+        .executable(
+            name: "TSCTestSupportExecutable",
+            targets: ["TSCTestSupportExecutable"]),
+    ],
+    // Rest of Package.swift truncated for brevity .....
+}
+```
+
+With the above Package descriptions, the `SwiftToolsSupport` product does not match a target within the package description it must be specified explicitly.
+
+```swift
+@package(url: "https://github.com/apple/swift-tools-support-core.git", products: ["SwiftToolsSupport"]) 
+import SwiftToolsSupport 
```

Comment:
This should say `import TSCBasic` since SwiftToolsSupport is not target.

Reply:
At one point in time, I wondered whether we should potentially synthesize a top-level module for every package which reexports all the root exported modules, so that users could always say `import Foo` given use of a package `Foo`.

If we had done such a thing, one might imagine the ability to depend on a product being optional here...